packages/react/src/lib/AutoControlledComponent.tsx

```diff
@@ -75,11 +75,20 @@ export default class AutoControlledComponent<P = {}, S = {}> extends UIComponent
   constructor(props, ctx) {
     super(props, ctx)
 
-    const { autoControlledProps } = this.constructor as any
+    const { autoControlledProps, getAutoControlledStateFromProps } = this.constructor as any
     const state = _.invoke(this, 'getInitialAutoControlledState', this.props) || {}
 
     if (process.env.NODE_ENV !== 'production') {
-      const { defaultProps, name, propTypes } = this.constructor as any
+      const { defaultProps, name, propTypes, getDerivedStateFromProps } = this.constructor as any
+
+      // require usage of getAutoControlledStateFromProps()
+      if (getDerivedStateFromProps !== AutoControlledComponent.getDerivedStateFromProps) {
+        /* eslint-disable-next-line no-console */
+        console.log(
```

Comment:
lets make it a error log - because this scenario will make a real issue to the client

Reply:
Overlooked this, great catch 👍